### PR TITLE
p2p: migrate to use new interface for channel errors

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: internal/p2p/p2ptest/require.go [hunk 1]
```diff
@@ -100,13 +101,18 @@ func RequireNoUpdates(ctx context.Context, t *testing.T, peerUpdates *p2p.PeerUp
...
+func RequireError(ctx context.Context, t *testing.T, channel *p2p.Channel, peerError p2p.PeerError) {
+	tctx, tcancel := context.WithCancel(ctx)
+	defer tcancel()
+
+	err := channel.SendError(tctx, peerError)
+	switch {
+	case errors.Is(err, context.DeadlineExceeded):
 		require.Fail(t, "timed out reporting error", "%v on %v", peerError, channel.ID)
+	case err == nil:
+		return
```
Comment (creachadair) [inline internal/p2p/p2ptest/require.go:113]: Harmless but redundant.

```suggestion
```

File: internal/p2p/channel.go [hunk 2]
```diff
@@ -61,10 +61,10 @@ func (pe PeerError) Unwrap() error { return pe.Err }
...
 // Each message is wrapped in an Envelope to specify its sender and receiver.
 type Channel struct {
-	ID    ChannelID
-	In    <-chan Envelope  // inbound messages (peers to reactors)
-	Out   chan<- Envelope  // outbound messages (reactors to peers)
-	Error chan<- PeerError // peer error reporting
+	ID  ChannelID
+	In  <-chan Envelope  // inbound messages (peers to reactors)
+	Out chan<- Envelope  // outbound messages (reactors to peers)
+	err chan<- PeerError // peer error reporting
```
Comment (creachadair) [inline internal/p2p/channel.go:67]: Consider maybe `errc` or `errCh` to avert confusion with the common use of `err` for error values (and we seem to use `xCh` as a convention for channels, for better or worse)

**Review by creachadair — APPROVED** [review]
👍🏼

**Review by cmwaters — APPROVED** [review]
LGTM. We still haven't worked out what peer errors really do right?

**Comment by tychoish** [comment]
> LGTM. We still haven't worked out what peer errors really do right?

It triggers this: https://github.com/tendermint/tendermint/blob/master/internal/p2p/router.go#L405-L412

Which causes the peer to be evicted (disconnected and then treated dis favorably for re-connection in the future.)

File: internal/p2p/p2ptest/require.go [hunk 3]
```diff
@@ -100,13 +101,16 @@ func RequireNoUpdates(ctx context.Context, t *testing.T, peerUpdates *p2p.PeerUp
 }
 
 // RequireError requires that the given peer error is submitted for a peer.
-func RequireError(t *testing.T, channel *p2p.Channel, peerError p2p.PeerError) {
-	timer := time.NewTimer(time.Second) // not time.After due to goroutine leaks
-	defer timer.Stop()
-	select {
-	case channel.Error <- peerError:
-	case <-timer.C:
+func RequireError(ctx context.Context, t *testing.T, channel *p2p.Channel, peerError p2p.PeerError) {
```
Comment (williambanfield) [inline internal/p2p/p2ptest/require.go:104]: Do the contexts sending to this function now set contexts with timeouts?

**Pull request merged**